jgo/util.py

```diff
@@ -0,0 +1,71 @@
+import psutil
+import sys
+
+from .jgo import jgo_main as main
+
+def add_jvm_args_as_necessary(argv, gc_option='-XX:+UseConcMarkSweepGC'):
+    """
+
+    Extend existing ``argv`` with reasonable default values for garbage collection and max heap size.
+    If ``-Xmx`` is not specified in ``argv``, set max heap size to half the system's memory.
+
+    :param argv: arugment vector
+    :param gc_option: Use this garbage collector settings, if any.
+    :return: ``argv`` with
+    """
+    if gc_option and not gc_option in argv:
+        argv += [gc_option]
+
+    for arg in argv:
+        if '-Xmx' in arg:
+            return argv
+
+    total_memory  = psutil.virtual_memory().total
+    exponent      = 3 if total_memory > 2*1024**3 else 2
+    memory_unit   = 'G' if exponent == 3 else 'M'
+    max_heap_size = '{}{}'.format(max(total_memory // (1024**exponent) // 2, 1), memory_unit)
+
+    argv = ['-Xmx{}'.format(max_heap_size)] + argv
+
+    return argv
+
+def maven_imagej_public_repo():
```

Comment:
A method like this is also present in scyjava_config; I changed it to `maven_scijava_repository`; see [here](https://github.com/scijava/scyjava/blob/8142d4117e448c881789a9ba553241fc5de9b215/scyjava_config.py#L35-L39). It's true that the URL is `maven.imagej.net` for now, but this will redirect to `maven.scijava.org` in the future.

Do you think it makes sense to have this repository shortcut in both packages? I don't feel strongly either way, but I do want to be consistent and call it `maven_scijava_repository` or similar, rather than `imagej`, since the repository is more general than only ImageJ.

Reply:
> A method like this is also present in scyjava_config; I changed it to `maven_scijava_repository`

Then we should probably do the same here.

> Do you think it makes sense to have this repository shortcut in both packages?

Although, technically, `jgo` does not need to no anything about the imagej/scijava repository, it is just very convenient for distributing applications via `jgo`. That's why I would like to have such a method in the `jgo` package.

As `scyjava` is downstream from `jgo`, we should probably just delegate to `jgo.maven_scijava_repository` if we decide to add it to `jgo`, too.

Reply:
Renamed `maven_imagej_public_repo` to `maven_scijava_repository` in 34aa66c